sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3162,6 +3162,9 @@ class Analyzer(override val catalogManager: CatalogManager)
       case WindowExpression(wf: FrameLessOffsetWindowFunction,
         WindowSpecDefinition(_, _, f: SpecifiedWindowFrame)) if wf.frame != f =>
         throw QueryCompilationErrors.cannotSpecifyWindowFrameError(wf.prettyName)
+      case WindowExpression(wf: AggregateWindowFunction, s: WindowSpecDefinition)
+        if wf.frame != UnspecifiedFrame =>
+        WindowExpression(wf, s.copy(frameSpecification = wf.frame))
```

Comment:
It seems duplicate with https://github.com/apache/spark/blob/a5af58bde544985d152bd974aa18cf0bdbfad2d5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L3171

Reply:
> It seems duplicate with
> 
> https://github.com/apache/spark/blob/a5af58bde544985d152bd974aa18cf0bdbfad2d5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L3171

Yea, but logically they are not same.
This behavior should be earlier before throw exception.

Reply:
This block is to use the default frame regardless of the specified frame ?

Reply:
> This block is to use the default frame regardless of the specified frame ?

Yea..PG do like this ==

Reply:
PG is not the golden standard Spark has to follow. To me, the PG behavior looks unreasonable as it silently ignores the user input. Can we investigate more databases? I'd expect that we should fail if a user specified window frame but the window function can only use its default frame.